user_tools/src/spark_rapids_pytools/rapids/tool_ctxt.py

```diff
@@ -131,9 +143,19 @@ def get_local_work_dir(self) -> str:
     def get_rapids_jar_url(self) -> str:
         # get the version from the package, instead of the yaml file
         # jar_version = self.get_value('sparkRapids', 'version')
-        mvn_base_url = self.get_value('sparkRapids', 'mvnUrl')
-        jar_version = Utils.get_latest_available_jar_version(mvn_base_url, Utils.get_base_release())
-        rapids_url = self.get_value('sparkRapids', 'repoUrl').format(mvn_base_url, jar_version, jar_version)
+        try:
```

Comment:
The point of the offline_mode is to avoid attempting requesting any resources over the web.
In this context  we could already know if the mode is offline or not simply by checking if the `resources/offline` folder contains any files/subdirectories.
So, we don't have to check the offline directory as a fallback; then set `self.set_ctxt('offline', True)`

Reply:
Updated the code